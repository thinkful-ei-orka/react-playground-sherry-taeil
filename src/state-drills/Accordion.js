import React from 'react';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
  }
  state = {
    clicked: null
  }
  handleClick = (index) => {
    this.setState({
      clicked: index
    });
  };
  render () {
    let sectionsLis = [];

    if (typeof this.props.sections === 'undefined') {
      sectionsLis.push(<li key="0"></li>);
    } else {
      this.props.sections.forEach((section, index) => {
        let paragraph = '';
        if (this.state.clicked === index) {
          paragraph = section.content;
        }
        sectionsLis.push(
          <li key={index}>
            <button onClick={() => this.handleClick(index)}>{section.title}</button>
            <p>{paragraph}</p>
          </li>
        );
      });
    }

    return (
      <div>
        <ul>
          {sectionsLis}
        </ul>
      </div>
    )
  }
}

export default Accordion
