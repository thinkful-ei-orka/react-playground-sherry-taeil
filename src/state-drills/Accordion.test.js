import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Accordion from './Accordion'
import AccordionData from './Accordion-data';

configure({ adapter: new Adapter() });

describe(`Accordion Snapshot Tests`, () => {
  it('Renders empty li when sections is not supplied', () => {
    expect(
      renderer.create(<Accordion />).toJSON()
    ).toMatchSnapshot()
  });

  it('Renders no sections as active by default', () => {
    expect(
      renderer.create(<Accordion sections={AccordionData.sections} />).toJSON()
    ).toMatchSnapshot()
  });

  it('Component opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={AccordionData.sections} />)
    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.text().includes('Lorem ipsum')).toBe(true);
  });

  it('Component only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections={AccordionData.sections} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')

    expect(wrapper.text().includes('Lorem ipsum')).toBe(false);
    expect(wrapper.text().includes('Cupiditate')).toBe(false);
    expect(wrapper.text().includes('Animi')).toBe(true);
  });
})
