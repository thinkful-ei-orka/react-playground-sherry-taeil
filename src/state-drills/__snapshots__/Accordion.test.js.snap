// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Snapshot Tests Renders empty li when sections is not supplied 1`] = `
<div>
  <ul>
    <li />
  </ul>
</div>
`;

exports[`Accordion Snapshot Tests Renders no sections as active by default 1`] = `
<div>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p>
        
      </p>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        
      </p>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p>
        
      </p>
    </li>
  </ul>
</div>
`;
